src/components/pages/Home/Home.tsx

```diff
@@ -1,19 +1,62 @@
 import SearchBar from '../../inputs/Search/SearchBar/SearchBar';
 import './styles.css';
+import { useStateValue } from '../../../state/state';
+import { useState, useEffect } from 'react';
+import AvailabilityHeader from '../../layout/AvailabilityHeader/AvailabilityHeader';
 
 function Home() {
+  const [{ arnsSourceContract }] = useStateValue();
+  const [records, setRecords] = useState({});
+
+  const [availability, setAvailability] = useState('search');
+  const [searchName, setSearchName] = useState('');
+
+  const [placeholderText, setPlaceholderText] = useState('Find a name');
+
+  useEffect(() => {
+    let newRecords = arnsSourceContract.records;
+    setRecords(newRecords);
+  }, [arnsSourceContract]);
+
+  function availiblityCheck(props: string) {
+    // if name is not in the legal character range or chars, dnt run
+    const namePattern = new RegExp('^[a-zA-Z0-9_-]{1,32}$');
+    if (!namePattern.test(props) && props !== '') {
+      console.log('not a valid name', namePattern.test(props));
+      return;
+    } else {
+      //if empty reset to search state
+      if (props == '') {
+        setAvailability('search');
+        setPlaceholderText('Find a name');
+      }
+      //if not registered set as available
+      if (!Object.keys(records).includes(props) && props !== '') {
+        setAvailability('available');
+        setSearchName(props);
+      }
+      //if registered set as unavailable
+      if (Object.keys(records).includes(props) && props !== '') {
+        setAvailability('unavailable');
+        setSearchName(props);
+        setPlaceholderText('try another name');
+      }
+    }
+  }
+
   return (
     <div className="page">
       <div className="pageHeader">Arweave Name System</div>
       <SearchBar
-        buttonAction={() => {
-          console.log('search');
-        }}
-        placeholderText="Enter a name"
-        headerText={'Find a domain name'}
+        buttonAction={(props) => availiblityCheck(props)}
+        placeholderText={placeholderText}
+        headerElement={
+          <AvailabilityHeader availability={availability} name={searchName} />
```

Comment:
remove `placeHolderText` from the SearchBar and pass it to AvailabilityHeader explicitly

Reply:
we can also have the `SearchBar` update the `availability` prop of this component instead of managing it in `Home` component

Reply:
placeholderText is specifically for the searchBar comp, the placeholder of the input in the searchBar, not sure what you mean for passing it to the header comp.